### PR TITLE
Usage insight links on questions and dashboards

### LINKED ISSUE #38474: Add a link to Metabase Analytics -> Question overview / Dashboard overview in the question/dashboard detail page
**Context**
Users may want to explore how a dashboard/question is used from the question/dashboard page. Today, the user must go to the Metabase analytics collection, navigate to the dashboard and filter the question/dashboard.

We can simply add a link in the question/dashboard detail page to the overview dashboard, filtering the content ID.

- Add a new option in the `...` menu to `Usage insights` (it should use the same Metabase analytics icon)
- If it is a question, this should lead to Question overview dashboard (entity-id `jm7KgY6IuS6pQjkBZ7WUI`) with a question_id=id param.
- If it is a dashboard, send to Dashboard overview dashboard (entity-id `bJEYb0o5CXlfWFcIztDwJ`) with dashboard_id=id

Other
- [ ] This link should appear only to saved questions and dashboards
- [ ] Only users with access to the Metabase analytics colllection can see this link (if this proves hard to build quickly, we can show only to admins and improve later)

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/38474 

### Description
This PR adds the `Usage Insights` option to the menus in Questions and Dashboards that will take you to the appropriate overview dashboard inside Instance Analytics for the given entity. Links will only appear in EE, and for users that have access to the Instance Analytics collection. 

### How to verify

1. As an Admin (Or a user with access to IA), go to any question
2. In the query builder header, click `...` and select "Usage Analytics". You should be brought to the Questions Overview dashboard filtered on that specific question you were just on
3. Try the same thing on Dashboards

### Demo
chrome_stTzOLeDgz

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by luizarakaki** [comment]
Nice stuff, working great.

**Review by luizarakaki — APPROVED** [review]

File: enterprise/frontend/src/metabase-enterprise/audit_app/components/InstanceAnalyticsButton/InstanceAnalyticsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const dispatch = useDispatch();
+  const entityId = useSelector(state =>
+    entitySelector(state as AuditInfoState),
+  );
+
+  useEffect(() => {
+    dispatch(loadInfo());
+  }, [dispatch]);
+
+  if (entityId !== undefined) {
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/audit_app/components/InstanceAnalyticsButton/InstanceAnalyticsButton.tsx:28]: I would maybe loosen up this check to `!entityId`

**Review by iethree — APPROVED** [review]
Very nice! 🏆

File: enterprise/frontend/src/metabase-enterprise/audit_app/components/InstanceAnalyticsButton/InstanceAnalyticsButton.tsx (hunk 1 above)
Comment (npfitz) [reply enterprise/frontend/src/metabase-enterprise/audit_app/components/InstanceAnalyticsButton/InstanceAnalyticsButton.tsx:28]: My fear here was IDs that are 0, though technically that shouldn't ever happen

File: enterprise/backend/src/metabase_enterprise/audit_app/api/user.clj [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+  "Gets audit info for the current user if he has permissions to access the audit collection.
+  Otherwise return an empty map."
+  []
+  (let [custom-reports     (audit-db/default-custom-reports-collection)
+        question-overview  (audit-db/entity-id->object :model/Dashboard audit-db/default-question-overview-entity-id)
+        dashboard-overview (audit-db/entity-id->object :model/Dashboard audit-db/default-dashboard-overview-entity-id)]
+    (cond-> {}
+      (mi/can-read? (audit-db/default-custom-reports-collection)) (assoc (:slug custom-reports) (:id custom-reports))
+      (mi/can-read? (audit-db/default-audit-collection)) (assoc (u/slugify (:name question-overview)) (:id question-overview)
+                                                                (u/slugify (:name dashboard-overview)) (:id dashboard-overview)))))
```
Comment (noahmoss) [inline enterprise/backend/src/metabase_enterprise/audit_app/api/user.clj:25]: nit — I find a `merge` with conditional args to be a little more readable in this type of situation, but up to you
```suggestion
    (merge
     {}
     (when (mi/can-read? (audit-db/default-custom-reports-collection))
       {(:slug custom-reports) (:id custom-reports)})
     (when (mi/can-read? (audit-db/default-audit-collection))
       {(u/slugify (:name question-overview)) (:id question-overview)
        (u/slugify (:name dashboard-overview)) (:id dashboard-overview)}))))
```

**Review by noahmoss — APPROVED** [review]

File: enterprise/frontend/src/metabase-enterprise/audit_app/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { createReducer } from "@reduxjs/toolkit";
+import { createAsyncThunk } from "metabase/lib/redux";
+
+import { GET } from "metabase/lib/api";
+import { isAuditInfoComplete } from "./selectors";
+import type { AuditInfoState } from "./types/state";
+
+const getAuditInfo = GET("/api/ee/audit-app/user/audit-info");
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/audit_app/reducer.ts:8]: oh, in hindsight I think i would have liked to see this in service.js or something to keep our API definitions centralized

Comment (npfitz) [reply enterprise/frontend/src/metabase-enterprise/audit_app/reducer.ts:8]: Good point. I'll follow this up tomorrow

**Pull request merged**